### PR TITLE
AWS - vpc NetworkLocation bugfix

### PR DESCRIPTION
**Pull request by anovis**
There is a case where the subnet or the security group could be `None` causing an error. I modified `process_resource` slightly to handle for this in the `NetworkLocation` class and added a test as well, which before would fail with the following error.

```UnboundLocalError: local variable 'sg_values' referenced before assignment
2020-06-08 17:28:47,651: custodian.commands:ERROR Error while executing policy test, continuing
Traceback (most recent call last):
  File "c7n\lib\site-packages\c7n\commands.py", line 280, in run
    policy()
  File "c7n\lib\site-packages\c7n\policy.py", line 1154, in __call__
    resources = mode.run()
  File "c7n\lib\site-packages\c7n\policy.py", line 290, in run
    resources = self.policy.resource_manager.resources()
  File "c7n\lib\site-packages\c7n\query.py", line 471, in resources
    resources = self.filter_resources(resources)
  File "c7n\lib\site-packages\c7n\manager.py", line 109, in filter_resources
    resources = f.process(resources, event)
  File "c7n\lib\site-packages\c7n\filters\vpc.py", line 179, in process
    found = self.process_resource(r, resource_sgs, resource_subnets, key)
  File "c7n\lib\site-packages\c7n\filters\vpc.py", line 245, in process_resource
    'security-groups': sg_values})
UnboundLocalError: local variable 'sg_values' referenced before assignment
```

### CODE REVIEW

**Comment by kapilt** [comment]
hi austen, long time, if your fixing an issue it would be helpful to have an issue/bug report (sample policy, version) associated.

**Comment by anovis** [comment]
hey! Its great to be contributing once again. 

the issue was on version 0.9.2 on the sample policy below where one of the alb did not have a security group attached. 

```
policies: 
  - name: network-location-test
    resource: aws.app-elb
    filters: 
      - type: network-location
        compare: 
          - subnet
          - security-group
        key: tag:NetworkLocation
```

**Comment by anovis** [comment]
also added bug report for tracking purposes.
#5892

File: tests/test_vpc.py [hunk 1]
```diff
@@ -534,6 +534,31 @@ def test_network_location_resource_missing(self):
                 "reason": "SecurityGroupMismatch"}],
         )
 
+    @functional
```
Comment (kapilt) [inline tests/test_vpc.py:537]: this shouldn't have a functional test decorator

**Pull request merged**